internal/client/tanzukubernetescluster/nodepool/nodepool_resource.go

```diff
@@ -0,0 +1,103 @@
+}
+
+// TanzuKubernetesNodePoolResourceServiceGet implements ClientService.
+func (c *Client) TanzuKubernetesNodePoolResourceServiceGet(fn *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolFullName) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error) {
+	queryParams := url.Values{}
+
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, fn.ManagementClusterName, provisioners, fn.ProvisionerName, tanzukubernetesclusters, fn.TanzuKubernetesClusterName, apiNodepoolsPath, fn.Name).AppendQueryParams(queryParams).String()
+	response := &tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse{}
+
+	err := c.Get(requestURL, response)
+
+	return response, err
+}
+
+// TanzuKubernetesNodePoolResourceServiceCreate implements ClientService.
+func (c *Client) TanzuKubernetesNodePoolResourceServiceCreate(request *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIRequest) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error) {
+	response := &tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse{}
+
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, request.Nodepool.FullName.ManagementClusterName, provisioners, request.Nodepool.FullName.ProvisionerName, tanzukubernetesclusters, request.Nodepool.FullName.TanzuKubernetesClusterName, apiNodepoolsPath).String()
+
+	err := c.Create(requestURL, request, response)
+
+	return response, err
+}
+
+// TanzuKubernetesNodePoolResourceServiceDelete implements ClientService.
+func (c *Client) TanzuKubernetesNodePoolResourceServiceDelete(fn *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolFullName) error {
+	queryParams := url.Values{}
+
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, fn.ManagementClusterName, provisioners, fn.ProvisionerName, tanzukubernetesclusters, fn.TanzuKubernetesClusterName, apiNodepoolsPath, fn.Name).AppendQueryParams(queryParams).String()
+
+	return c.Delete(requestURL)
+}
+
+// TanzuKubernetesNodePoolResourceServiceList implements ClientService.
+func (c *Client) TanzuKubernetesNodePoolResourceServiceList(cluster *tkcmodels.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterFullName) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolListNodepoolsResponse, error) {
+	queryParams := url.Values{}
+
+	// for stability of the results
+	queryParams.Add(queryParamKeySortBy, "createTime")
```

Comment:
Currently, backend doesn't support _sortBy_.

Reply:
Okay, thanks for that, I'll remove it. 
Since the query sortBy exists in the API I took a similar implementation from one of the other nodepool tmc resources.